### PR TITLE
UG update: Move `Viewing Clients`, `Finding Clients`, `Updating Clients` Sections to UG

### LINKED ISSUE #22: UG update: Move "Viewing Clients", "Finding Clients", "Updating Clients" Sections to UG

### PR DESCRIPTION
**Pull request by rtshkmr**
Fixes #22

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -83,46 +84,54 @@ Examples:
...
-Format: `edit INDEX [n/NAME] [p/PHONE] [e/EMAIL] [a/ADDRESS] [t/TAG]…​`
+### Finding clients: `client find`
 
-* Edits the person at the specified `INDEX`. The index refers to the index number shown in the displayed person list. The index **must be a positive integer** 1, 2, 3, …​
-* At least one of the optional fields must be provided.
-* Existing values will be updated to the input values.
-* When editing tags, the existing tags of the person will be removed i.e adding of tags is not cumulative.
-* You can remove all the person’s tags by typing `t/` without
-    specifying any tags after it.
+Searches for clients that matches the given name.
```
Comment (qwoprocks) [inline docs/UserGuide.md]: Might want to change this to match the original "Finds clients whose names contain any of the given keywords.", and change `n/NAME` to `KEYWORD [MORE_KEYWORDS]` as the format section below mentions keywords

**Review by qwoprocks — CHANGES_REQUESTED** [review]

File: docs/UserGuide.md (hunk 1 above)
Comment (rtshkmr) [reply docs/UserGuide.md]: alright, originally copied from our hackmd document, which seems to be out of date based on what you pointed out
image

File: docs/UserGuide.md [hunk 2]
```diff
@@ -83,46 +83,56 @@ Examples:
...
 
-Examples:
-*  `edit 1 p/91234567 e/johndoe@example.com` Edits the phone number and email address of the 1st person to be `91234567` and `johndoe@example.com` respectively.
-*  `edit 2 n/Betsy Crower t/` Edits the name of the 2nd person to be `Betsy Crower` and clears all existing tags.
+Format: `client find KEYWORD [MORE_KEYWORDS]`
+* The search is case-insensitive. e.g hans will match Hans
+* The order of the keywords does not matter. e.g. Hans Bo will match Bo Hans
+* Only the name is searched.
+* Only full words will be matched e.g. Han will not match Hans
+* Persons matching at least one keyword will be returned (i.e. OR search). e.g. Hans Bo will return Hans Gruber, Bo Yang
```
Comment (qwoprocks) [inline docs/UserGuide.md]:
```suggestion
* The search is case-insensitive. e.g `hans` will match `Hans`
* The order of the keywords does not matter. e.g. `Hans Bo` will match `Bo Hans`
* Only the name and country are searched.
* Only full words will be matched e.g. `Han` will not match `Hans`
* Persons matching at least one keyword will be returned (i.e. OR search). e.g. `Hans Bo Russia` will return `Hans Gruber`, `Bo Yang`, `Alice Katya`
```

File: docs/UserGuide.md [hunk 3]
```diff
@@ -83,46 +83,56 @@ Examples:
...
+
+Examples: 
+
+* `client update 5 n/Katya` Only updates **name**, other fields remain the same
+* `client update 4 n/Alek p/34842097 e/dcsdcr@nus.edu.sg` Updates **name**, **phone number** and **email**, other fields remain the same.
+
+
+### Listing all persons : `list`
+
+Shows a list of all persons in the address book.
```
Comment (qwoprocks) [inline docs/UserGuide.md]:
```suggestion
### Listing all clients: `list`

Shows a list of all clients in the address book.
```

**Review by qwoprocks — CHANGES_REQUESTED** [review]
Small changes

File: docs/UserGuide.md [hunk 4]
```diff
@@ -83,46 +83,56 @@ Examples:
...
-Format: `edit INDEX [n/NAME] [p/PHONE] [e/EMAIL] [a/ADDRESS] [t/TAG]…​`
+### Finding clients: `client find`
 
-* Edits the person at the specified `INDEX`. The index refers to the index number shown in the displayed person list. The index **must be a positive integer** 1, 2, 3, …​
-* At least one of the optional fields must be provided.
-* Existing values will be updated to the input values.
-* When editing tags, the existing tags of the person will be removed i.e adding of tags is not cumulative.
-* You can remove all the person’s tags by typing `t/` without
-    specifying any tags after it.
+Finds clients whose names contain any of the given keywords.
```
Comment (qwoprocks) [inline docs/UserGuide.md]:
```suggestion
Finds clients whose names contain any of the given keywords, or whose country of residence contains any of the given keywords.
```

File: docs/UserGuide.md (hunk 3 above)
Comment (tankangliang) [reply docs/UserGuide.md]: Should be added to Table of Contents as well if we're adding in a new command `list`

Comment (rtshkmr) [reply docs/UserGuide.md]: `list` is an AB3 command, I didn't add that in actually. Instead, I copied over from our HackMD file what I had to add in.
that's why `person` persisted and wasn't changed to `clients`

I'll just remove it and update this PR

**Review by qwoprocks — APPROVED** [review]

**Pull request merged**